pkg/workspace_template.go

```diff
@@ -213,29 +219,11 @@ func generateArguments(spec *WorkspaceSpec, config SystemConfig, withRuntimeVars
 		Type:  "input.hidden",
 	})
 
-	if withRuntimeVars {
-		// Host
-		systemParameters = append(systemParameters, Parameter{
-			Name:  "sys-host",
-			Value: config.Domain(),
-			Type:  "input.hidden",
-		})
-
-		// Node pool parameter and options
-		var options []*ParameterOption
-		if err = yaml.Unmarshal([]byte(config["applicationNodePoolOptions"]), &options); err != nil {
-			return
-		}
-		systemParameters = append(systemParameters, Parameter{
-			Name:        "sys-node-pool",
-			Value:       ptr.String(options[0].Value),
-			Type:        "select.select",
-			Options:     options,
-			DisplayName: ptr.String("Node pool"),
-			Hint:        ptr.String("Name of node pool or group"),
-			Required:    true,
-		})
+	runtimeParameters, err := generateRuntimeParameters(config)
```

Comment:
Inject runtime parameters here so Workflow validations passes and we don't have inject the other parts of the manifest.